ouroboros-network-framework/src/Ouroboros/Network/ConnectionManager/Core.hs

```diff
@@ -682,8 +687,7 @@ withConnectionManager ConnectionManagerArguments {
             -- using 'cancel' here, since we want to block until connection
             -- handler thread terminates.
             traverse_ cancel (getConnThread connState)
-          )
-          state
+          ))
```

Comment:
would it work if we `throwTo` and then wait for all the threads to terminate; this would avoid calling `cancel` from a separate thread.  The drawback is that if a connection is executing the `cleanup` handler, it will block for the `TIME_WAIT` interval.

Reply:
`throwTo` is also possibly blocking right? waiting on the thread to actually receive the exception. So I do not think we would gain much from using `throwTo`.

Reply:
@coot

Reply:
* `cancel` blocks until thread terminates
* `throwTo` blocks until an exception is raised

so `cancel` can block for much longer than `throwTo`, since during `TIME_WAIT` interval, we will accept asynchronous exceptions (though they will be ignored).  This will unblock `throwTo`, but not `cancel`.

Reply:
Addressed